### ISSUE TITLE
example base image links to deprecated script

### ISSUE THREAD

**Issue by szeitlin** [docker/docker.github.io#3537]
File: [engine/userguide/eng-image/baseimages.md](https://docs.docker.com/engine/userguide/eng-image/baseimages/), CC @mstanleyjones

This section on creating base images could use some updating. For example, this link to the Debian/Ubuntu example: https://github.com/moby/moby/blob/master/contrib/mkimage-debootstrap.sh was last updated i n2015, and includes a line that prints:

"warning: this script is deprecated - see mkimage.sh and mkimage/debootstrap"

FWIW, I was looking for information on how to take an existing base image and customize it - what to include, what not to include, for a project that is still under construction. 

I'm trying to solve the problem of having to rebuild too much of the container each time, because my base image is just ubuntu and installing all the packages isn't really necessary every time I update any tiny piece of my other code. It's time-consuming to the point where I'm afraid my other teammates will refuse to use Docker. 

So far I'm not finding much in the way of best practices for how to turn an image I've already built into a base image? Is there not a way to do that?

**Comment by joaofnfernandes** [comment]
I'm not 100% sure I understand what you mean. It seems your case is something like

```
[ubuntu] <--- [your image]
```

And basically in the Dockerfile of your image, you're installing Ubuntu packages and then running you app. So every time you make changes to your code and need to rebuild the image, you also need to install packages, which takes time.

If this is your case, and you're trying to find out how to update that ubuntu base image to have the packages pre-installed, you can do something like this

```
[ubuntu] <-- [your ubuntu with packages] <-- [your app]
```

You can create an image based on Ubuntu and installs all the packages that you need. Then, you create a new image for your app. This image is based on the one you've created that has the packages installed, instead of Ubuntu directly.

This will allow you to develop your app without having to fetch and install packages every time, and also allows you to update the intermediate image and add more packages if you ever need it.

Let me know if I totally misread your question and what I'm suggesting wouldn't work for you.

**Comment by szeitlin** [comment]
thanks yeah I figured out how to do it. I just didn't see it documented anywhere. 

So I ended up with two Dockerfiles. The first one modifies the ubuntu base image to install the other dependencies I need, and makes a new 'base image'. 

Then I do `FROM baseimage` in my 2nd Dockerfile, which lives in a child directory of the first.

That seems to work beautifully. It just wasn't clear to me that this approach would work.
